Easydict/Swift/Feature/Configuration/Configuration.swift

```diff
@@ -419,6 +424,37 @@ class Configuration: NSObject {
                     self?.didSetAppearance(newValue)
                 }
         )
+        cancellables.append(
+            Defaults.publisher(.pinShortcut)
+                .removeDuplicates()
+                .sink { [weak self] _ in
+                    self?.updateWindowTitlebar()
+                }
+        )
+
+        cancellables.append(
+            Defaults.publisher(.appleDictionaryShortcut)
+                .removeDuplicates()
+                .sink { [weak self] _ in
+                    self?.updateWindowTitlebar()
+                }
+        )
+
+        cancellables.append(
+            Defaults.publisher(.googleShortcut)
+                .removeDuplicates()
+                .sink { [weak self] _ in
+                    self?.updateWindowTitlebar()
+                }
+        )
+
+        cancellables.append(
+            Defaults.publisher(.eudicShortcut)
+                .removeDuplicates()
+                .sink { [weak self] _ in
+                    self?.updateWindowTitlebar()
+                }
+        )
```

Comment:
```suggestion
         Publishers.Merge4(
            Defaults.publisher(.pinShortcut),
            Defaults.publisher(.appleDictionaryShortcut),
            Defaults.publisher(.googleShortcut),
            Defaults.publisher(.eudicShortcut)
        )
        .removeDuplicates()
        .sink { [weak self] _ in
            self?.updateWindowTitlebar()
        }
        .store(in: &cancellables)
```

Too much duplicate code, can we optimize it like this?

Reply:
Also, I noticed that `cancellables` is an Array type instead of a Set, why?

```swift
var cancellables: [AnyCancellable] = []
```

Reply:
Can we use Set here? `store(in:` looks more concise than the nested `append`.

Reply:
Also, I noticed that `cancellables` is an Array type instead of a Set, why?

```swift
var cancellables: [AnyCancellable] = []
```

Can we use Set here? `store(in:` looks more concise than the nested `append`.